rust-doc-style-guide

```diff
@@ -0,0 +1,268 @@
+
+  - Use `bash` syntax highlighting
+
+- Rust declarations: commands, functions, arguments, parameters, flags, variables
+- In-line command line output
+
+  > Writing a program that prints `Hello, world!`
+
+- Data types: `i8`, `u128`
+- Names of crates, traits, libraries,
+- Command line tools, plugins, packages
+- Files: `config.toml`
+- Architectures and targets
+
+ > The `RISC-V` and `Xtensa` architectures.
+
+### Monospace and Other Types of Formatting
+
+Monospace font can also be used in:
+
+- links
+
+  > [`String`](https://doc.rust-lang.org/std/string/struct.String.html) is a string type provided by ...
+
+- headings
+
+  > ### A `no_std` Rust Environment
+
+
+## Using _Italics_
+
+- Introduce new terms
+
+  > Each value in Rust has an _owner_.
+  > A _hash map_ allows you to associate a value with a particular key.
+
+- Emphasize important concepts or words
+
+  > When `s` comes _into_ scope, it is valid. It remains valid until it goes _out of_ scope.
+
+
+## Mode of Narration
+
+- Use _the first person_ (we) when introducing a tutorial or explaining how things will be done. The reader will feel like being on the same team with the authors working side by side.
+
+  > We'll start writing a program for the LM3S6965, a Cortex-M3 microcontroller. We have chosen this as our initial target because it can be emulated using QEMU so you don't need to fiddle with hardware in this section, and we can focus on the tooling and the development process.
+
+- Use _the second person_ (you) when describing what the reader should do while installing software, following a tutorial or a procedure. However, in most cases you can use imperative mood as if giving orders to the readers. It makes instructions much shorter and clearer.
+
+  > 1\. Install `cargo-generate`
+  >
+  > `cargo install cargo-generate`
+  >
+  > 2\. Generate a new project
+  >
+  > `cargo generate --git https://github.com/rust-embedded/cortex-m-quickstart`
+
+- Use _the third person_ (the user, it) when describing how things work from the perspective of hardware or software
+
+  > A driver has to be initialized with an instance of type that implements a certain `trait` of the embedded-hal which is ensured via trait bound and provides its own type instance with a custom set of methods allowing to interact with the driven device.
+
+
+## Terminology
+
+The content here is split into two categories:
+
+- **Recommended terms** covers which terms are better to use in which circumstances
+- **Spelling issues** lists term spellings chosen by the team
+
+### Recommended Terms
+
+- _SoC_ and _chip_ -- these terms are used interchangeably. Should only one be used?
+- _Product_ -- suggested as an umbrella term for Espressif chips, modules, development boards, etc. Otherwise, you need to list all product types to avoid ambiguities.
+
+### Spelling issues
+
+- _ESP-IDF_ or _esp-idf_?
+- Cargo or cargo or `cargo`?
```

Comment:
These are some ideas for the terms to be standardized. Need to decide on their spelling.

You will probably include more terms like this in the future.